plaid/src/data/okta/mod.rs

```diff
@@ -92,63 +114,103 @@ impl Okta {
-                        error!("Got an invalid date from Okta: {}", log.published);
+                        error!("Got an invalid date from Okta: {}", published);
                         continue;
                     }
                 };
 
-                if newest_timestamp.is_none() || log_timestamp > newest_timestamp.unwrap() {
-                    newest_timestamp = Some(log_timestamp);
-                }
-
-                if oldest_timestamp.is_none() || log_timestamp < oldest_timestamp.unwrap() {
-                    oldest_timestamp = Some(log_timestamp);
+                // Check if this is the latest log we've seen and update if so
+                // We'll use the new most_recent_log_seen time to filter the subsequent
+                // API calls to Okta afterwards
+                //
+                // By default, the Okta API returns the logs in ascending order so we could in theory just
+                // take the last timstamp and set it as our max log time. I'm okay with doing another check here in the
+                // case that Okta's sorting fails to ensure that we do not miss any logs. The number of comparisions here (1000 max)
+                // is nothing to be concerned about.
+                if log_timestamp > most_recent_log_seen {
+                    most_recent_log_seen = log_timestamp;
                 }
 
-                if log_timestamp < self.since {
-                    self.since = newest_timestamp.unwrap();
-                    return Ok(());
-                }
+                // Attempts to parse the log received from Okta to bytes.
+                let log_bytes = match serde_json::to_vec(&log) {
+                    Ok(bytes) => bytes,
+                    Err(e) => {
+                        error!("Failed to serialize Okta logs to bytes. Error: {e}");
+                        continue;
+                    }
+                };
 
-                counter += 1;
-                if log_timestamp > self.since {
-                    // Eventually these errors need to bubble up so the service can shut down
-                    // then be restarted by an orchestration service
-                    self.logger
-                        .send(Message::new(
-                            format!("okta"),
-                            serde_json::to_string(&log).unwrap().as_bytes().to_vec(),
-                            LogSource::Generator(Generator::Okta),
-                            self.config.logbacks_allowed.clone(),
-                        ))
-                        .unwrap();
-                }
+                // Send log into logging system to be processed by rule(s)
+                //
+                // Eventually these errors need to bubble up so the service can shut down
+                // then be restarted by an orchestration service
+                self.logger
+                    .send(Message::new(
+                        "okta".to_string(),
+                        log_bytes,
```

Comment:
I think this should be serializing out the Value to a string then getting the bytes of that